src/prolong_restrict/pr_ops.hpp

```diff
@@ -157,9 +157,11 @@ struct RestrictAverage {
     const Real tvol = ((vol[0][0][0] + vol[0][1][0]) + (vol[0][0][1] + vol[0][1][1])) +
                       ((vol[1][0][0] + vol[1][1][0]) + (vol[1][0][1] + vol[1][1][1]));
     coarse(element_idx, l, m, n, ck, cj, ci) =
-        (((terms[0][0][0] + terms[0][1][0]) + (terms[0][0][1] + terms[0][1][1])) +
-         ((terms[1][0][0] + terms[1][1][0]) + (terms[1][0][1] + terms[1][1][1]))) /
-        tvol;
+        tvol > 0.0
+            ? (((terms[0][0][0] + terms[0][1][0]) + (terms[0][0][1] + terms[0][1][1])) +
+               ((terms[1][0][0] + terms[1][1][0]) + (terms[1][0][1] + terms[1][1][1]))) /
+                  tvol
+            : 0.0;
```

Comment:
This is the only real fix. Previously a face at the origin would have `tvol = 0.0`, so restricting things like the diffusion coefficients in the solvers produced NaNs when not using Cartesian coords.

Reply:
would it be better to express this as a masked term? I guess for prolongation/restriction it probably doesn't matter

Reply:
Do you mean just wrap it in an if-else or some other way of masking?

Reply:
I meant this, but I dunno if it's any better
```suggestion
        (tvol > 0.0)*
            (((terms[0][0][0] + terms[0][1][0]) + (terms[0][0][1] + terms[0][1][1])) +
               ((terms[1][0][0] + terms[1][1][0]) + (terms[1][0][1] + terms[1][1][1]))) /
                  (tvol + 1e-20);
```

Reply:
Yeah, I doubt it makes much difference here.